### PR TITLE
Add CHANGELOG entry for #17654

### PR DESCRIPTION
**Pull request by kamipo**
Add CHANGELOG entry for #17654.

### CODE REVIEW

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+
+        # database.yml
+        production:
+          adapter: mysql2
+          database: foo_prod
+          user: foo
+          strict: :default
+
+    *Ryuta Kamizono*
+
```
Comment (eileencodes) [inline activerecord/CHANGELOG.md]: Thanks for the changelog entry! The grammar is a little off and is confusing to read. Can you update the language to say:

"Do not set `sql_mode` if `strict: :default` is specified"

And then put the database.yml example in back ticks so it's a code block. Like

```
# database.yml
  production:
  adapter: mysql2
  database: foo_prod
  user: foo
  strict: :default
```

**Comment by kamipo** [comment]
Thank you for the review! I fixed it!

**Comment by eileencodes** [comment]
Thanks! :smile:

**Comment by kamipo** [comment]
:smile:

**Pull request merged**